Content.Server/StationEvents/Metric/StationMetricSystem.cs

```diff
@@ -0,0 +1,97 @@
+using Content.Server.Construction.Completions;
+using Content.Server.GameTicking;
+using Content.Server.StationEvents.Metric.Components;
+using Content.Shared.chaos;
+using Robust.Shared.Map;
+using Robust.Shared.Prototypes;
+
+namespace Content.Server.StationEvents.Metric;
+
+public class StationMetricSystem : EntitySystem
+{
+    [Dependency] private readonly IEntityManager _entities = default!;
+    [Dependency] private readonly IPrototypeManager _prototype = default!;
+    public override void Initialize()
+    {
+        base.Initialize();
+        //SubscribeLocalEvent<T, CalculateChaosEvent>(OnCalculateChaos);
+    }
+
+    public Dictionary<EntityPrototype, ChaosMetricComponent> AllMetrics()
+    {
+        var allMetrics = new Dictionary<EntityPrototype, ChaosMetricComponent>();
+        foreach (var prototype in _prototype.EnumeratePrototypes<EntityPrototype>())
+        {
+            if (prototype.Abstract)
+                continue;
+
+            if (!prototype.TryGetComponent<ChaosMetricComponent>(out var metric))
+                continue;
+
+            allMetrics.Add(prototype, metric);
+        }
```

Comment:
Please cache this.

Reply:
It's intended to be a one-off.

Reply:
If we did cache this, it would mean you can't hot-reload metrics changes. Right now you can change the metrics definitions, then endgamerule, addgamerule the director and it will pick them up. If there was a cache, it would retain those changes and prevent hot reloading.

There's not actually a performance problem here, it only picks up the metrics when the rule starts. So in a typical round a cache would use memory without making any difference.